elastic/apm-agent-nodejs
# ci(jenkins): run integrations tests for the nodejs agent only

Relates to https://github.com/elastic/apm-integration-testing/issues/41

## Highlights
- It's triggered for each PR by default and disabled to any upstream branches.
- Trigger the Integration Tests, aka ITs, pipelines from https://github.com/elastic/apm-integration-testing based on PRs or on demand, aka manually.
- It does trigger only the `Node.js` stage in the ITs.
- ITs should be async.
- This new stage notifies the GitHub check with the status of the execution.

## How to test it out locally

```
python scripts/compose.py start 7.0.0 \
 --nodejs-agent-package=v1v/apm-agent-nodejs#bd5820e7fc06229ffd854adcb39f2655223a590c \
 --with-agent-nodejs-express \
 --no-apm-server-dashboards --no-apm-server-self-instrument --no-kibana
```

```
 docker run --rm -ti apm-integration-testing_agent-nodejs-express /bin/bash
root@f4939ca306a2:/app# npm install v1v/apm-agent-nodejs#bd5820e7fc06229ffd854adcb39f2655223a590c
npm WARN saveError ENOENT: no such file or directory, open '/app/package.json'
npm notice created a lockfile as package-lock.json. You should commit this file.
npm WARN enoent ENOENT: no such file or directory, open '/app/package.json'
npm WARN app No description
npm WARN app No repository field.
npm WARN app No README data
npm WARN app No license field.

+ elastic-apm-node@2.11.6
added 106 packages from 95 contributors and audited 168 packages in 6.309s
found 0 vulnerabilities

root@f4939ca306a2:/app# cat package-lock.json  | grep apm-node -A 6
    "elastic-apm-node": {
      "version": "github:v1v/apm-agent-nodejs#bd5820e7fc06229ffd854adcb39f2655223a590c",
      "from": "github:v1v/apm-agent-nodejs#bd5820e7fc06229ffd854adcb39f2655223a59

```
